libcudacxx/include/cuda/std/__type_traits/is_convertible.h

```diff
@@ -90,121 +76,66 @@ _CCCL_END_NV_DIAG_SUPPRESS()
 _CCCL_DIAG_POP
 
 template <class _From, class _To, class = void>
-struct __is_convertible_test : public false_type
-{};
+inline constexpr bool __is_convertible_test_v = false;
 
 template <class _From, class _To>
-struct __is_convertible_test<
+inline constexpr bool __is_convertible_test_v<
   _From,
   _To,
-  decltype(::cuda::std::__is_convertible_imp::__test_convert<_To>(::cuda::std::declval<_From>()))> : public true_type
-{};
+  decltype(::cuda::std::__is_convertible_imp::__test_convert<_To>(::cuda::std::declval<_From>()))> = true;
 
 template <class _Tp, bool _IsArray = is_array_v<_Tp>, bool _IsFunction = is_function_v<_Tp>, bool _IsVoid = is_void_v<_Tp>>
-struct __is_array_function_or_void
-{
-  enum
-  {
-    value = 0
-  };
-};
+inline constexpr int __is_array_function_or_void_v = 0;
```

Comment:
Nitpick: If we are touching this we should turn those arbitrary numbers into enumerations